src/open_r1/rewards.py

```diff
@@ -379,6 +381,42 @@ def evaluate_code(code, test_cases):
     return rewards
 
 
+def ioi_code_reward(completions, test_batch_size: int = 1, **kwargs) -> list[float]:
+    """Reward function that evaluates IOI problems using Piston+our IOI package.
+    
+    Assumes the dataset has the same format as hf.co/datasets/open-r1/ioi
+
+    test_batch_size: evaluate these many test cases in parallel, then check if any of them failed (0 score): if so stop evaluating; otherwise continue with the next batch of test cases.
+    """
+    # for info on setting up piston workers, see slurm/piston/README.md
+    piston_client = get_piston_client_from_env()
+    
+    code_snippets = [
+        # note: grading is automatically skipped if no code is extracted
+        add_includes(extract_code(completion[-1]["content"], 'cpp'), problem_id) 
+        for completion, problem_id in zip(completions, kwargs["id"])
+    ]
+
+    async def run_catch_exceptions(task):
+        try:
+            return await task
+        except Exception as e:
+            print(f"Error from Piston worker: {e}")
+            return SubtaskResult()  # score 0.0
+
+    # load problem data. undo separating kwargs by column
+    problems_data = [dict(zip(kwargs.keys(), values)) for values in zip(*kwargs.values())]
+    
+    loop = _init_event_loop()
+    evals = [
+        loop.create_task(run_catch_exceptions(score_subtask(piston_client, problem_data, code, test_batch_size=test_batch_size)))
```

Comment:
```suggestion
        asyncio.create_task(run_catch_exceptions(score_subtask(piston_client, problem_data, code, test_batch_size=test_batch_size)))
```

Reply:
yeah this works because we set the asyncio event loop but you do need to keep the _init line otherwise you'll get an error from this